File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
    let str = "";
    let sk = [];
    for (let i = 0; i < s.length; i++) {
        if (s[i] === '(') {
            if (sk.length !== 0) str += '(';
            sk.push(s[i]);
        } else {
            sk.pop();
            if (sk.length !== 0) str += ')';
        }
    }
    return str;
};

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        String str = "";
        Stack<Character> sk = new Stack<>();
        for (char c : s.toCharArray()) {
            if (c == '(') {
                if (sk.size() != 0) str += '(';
                sk.push(c);
            } else {
                sk.pop();
                if (sk.size() != 0) str += ')';
            }
        }
        return str;
    }
}